Source/WebCore/xml/XSLImportRule.cpp

```diff
@@ -73,27 +69,24 @@ bool XSLImportRule::isLoading()
 
 void XSLImportRule::loadSheet()
 {
-    CachedResourceLoader* cachedResourceLoader = nullptr;
-
-    XSLStyleSheet* rootSheet = parentStyleSheet();
+    RefPtr rootSheet = parentStyleSheet();
 
     if (rootSheet) {
-        while (XSLStyleSheet* parentSheet = rootSheet->parentStyleSheet())
+        while (auto* parentSheet = rootSheet->parentStyleSheet())
             rootSheet = parentSheet;
     }
 
-    if (rootSheet)
-        cachedResourceLoader = rootSheet->cachedResourceLoader();
+    RefPtr cachedResourceLoader = rootSheet->cachedResourceLoader();
```

Comment:
The code just above this checks `rootSheet` for null, but here we dereference the pointer unconditionally. The old code didn’t do that.

Reply:
Yes, this is true but line 99/106 unconditionally dereferenced the resource loader.

Whilst it's after an early exit for a cyclic relationship, I don't think that's relevant as far as nullify of the resource loader is concerned.

Reply:
We could remove the unnecessary null check above then?

Reply:
I suppose we could. I'll go do that.